### PR TITLE
ci(cypress): Fix card expiry for savecard flows

### PR DESCRIPTION
**Pull request by Gnanasundari24**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description

-Resolved an issue in the savecard flow where card expiry was incorrectly sourced from the confirmBody.json, resulting in failures. Now, only the payment_token is utilized, ensuring successful payment transactions.
-This pull request includes refactoring of the card_pm details to accommodate the addition of banks and wallet flows.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

To enhance test coverage for wallet and bank redirect flows.

## How did you test it?

Ran the test manually
Screenshot 2024-05-08 at 3 56 21 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by likhinbopanna — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

File: cypress-tests/cypress/support/commands.js [hunk 1]
```diff
@@ -295,20 +295,20 @@ Cypress.Commands.add("createConfirmPaymentTest", (createConfirmPaymentBody, deta
 });
 
 // This is consequent saved card payment confirm call test(Using payment token)
-Cypress.Commands.add("saveCardConfirmCallTest", (confirmBody, det, globalState) => {
+Cypress.Commands.add("saveCardConfirmCallTest", (SaveCardConfirmBody,det,globalState) => {
```
Comment (SanchithHegde) [inline cypress-tests/cypress/support/commands.js:298]: Nit: variable naming, can be taken in a separate PR.

```suggestion
Cypress.Commands.add("saveCardConfirmCallTest", (saveCardConfirmBody,det,globalState) => {
```

**Review by SanchithHegde — APPROVED** [review]
